daemon/cmd/daemon.go

```diff
@@ -288,7 +288,15 @@ func NewDaemon(ctx context.Context, dp datapath.Datapath) (*Daemon, *endpointRes
 		return nil, nil, fmt.Errorf("unable to setup encryption: %s", err)
 	}
 
-	mtuConfig := mtu.NewConfiguration(authKeySize, option.Config.EnableIPSec, option.Config.Tunnel != option.TunnelDisabled, configuredMTU)
+	var mtuConfig mtu.Configuration
+	if node.GetExternalIPv4() != nil {
+		mtuConfig = mtu.NewConfiguration(authKeySize, option.Config.EnableIPSec, option.Config.Tunnel != option.TunnelDisabled, configuredMTU, node.GetExternalIPv4())
+	} else if node.GetIPv6() != nil {
+		mtuConfig = mtu.NewConfiguration(authKeySize, option.Config.EnableIPSec, option.Config.Tunnel != option.TunnelDisabled, configuredMTU, node.GetIPv6())
+	} else {
+		log.Warn("Nodes don't have an external IP. MTU will be fetched from the default interface")
+		mtuConfig = mtu.NewConfiguration(authKeySize, option.Config.EnableIPSec, option.Config.Tunnel != option.TunnelDisabled, configuredMTU, nil)
+	}
```

Comment:
nit, this is probably a little simpler with:
```suggestion
	externalIP := node.GetExternalIPv4() 
	if externalIP == nil {
		externalIP = node.GetIPv6()
	}
	mtuConfig := mtu.NewConfiguration(authKeySize, option.Config.EnableIPSec, option.Config.Tunnel != option.TunnelDisabled, configuredMTU, externalIP)
```

Then the extra log warning here about not having external IP will already be implied by the other log messages in the path taken inside `mtu.NewConfiguration()`.

Reply:
ok, I'll add a comment to clarify this